### PR TITLE
OPHJOD-1123: Unify opportunity pages

### PR DESCRIPTION
**Pull request by juhaniko**
## Description

* Created a generic page for opportunities and refactored job and education routes to use it
* Fix a bug in favorites: All favorite links pointed to job opportunity route regardless of the type

## Related JIRA ticket

https://jira.eduuni.fi/browse/OPHJOD-1123

### CODE REVIEW

File: src/routes/Profile/Favorites/Favorites.tsx [hunk 1]
```diff
@@ -154,7 +157,7 @@ const Favorites = () => {
           return (
             <OpportunityCard
               key={id}
-              to={`/${language}/${t('slugs.job-opportunity.index')}/${id}`}
+              to={`/${language}/${getTypeSlug(mahdollisuusTyyppi)}/${id}`}
```
Comment (juhaniko) [inline src/routes/Profile/Favorites/Favorites.tsx:160]: Noticed this bug by chance and decided to fix it although it wasn't part of the ticket. The card was trying to route to job opportunity page even if favorite type was `KOULUTUSMAHDOLLISUUS`

File: src/i18n/en/translation.json
Comment (juhaniko) [inline src/i18n/en/translation.json:1]: I updated the job and education opportunity slugs to match fi file and also sorted this JSON

File: src/routes/JobOpportunity/JobOpportunity.tsx [hunk 2]
```diff
@@ -1,44 +1,17 @@
...
-}
-const ScrollHeading = ({ title, heading, className }: ScrollHeadingProps) => {
-  const HeadingTag = heading as keyof JSX.IntrinsicElements;
-  return (
-    <HeadingTag id={title} className={tc(`scroll-mt-[96px] ${className}`)}>
-      {title}
-    </HeadingTag>
-  );
-};
+import { type LoaderData } from './loader';
```
Comment (ketsappi) [inline src/routes/JobOpportunity/JobOpportunity.tsx:9]: Is it necessary to explicitly tell the `type` here?

File: src/components/OpportunityDetails/OpportunityDetails.tsx [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+import { useToolStore } from '@/stores/useToolStore';
+import { getLocalizedText } from '@/utils';
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import { MdCompareArrows, MdOutlinePrint, MdOutlineRoute, MdOutlineShare } from 'react-icons/md';
+
+export interface OpportunityDetailsSection {
+  navTitle: string; // Navigation title
+  content: JSX.Element; // Section content
+  dev?: boolean; // Only render in dev environment
```
Comment (ketsappi) [inline src/components/OpportunityDetails/OpportunityDetails.tsx:23]: These properties are quite self explanating, no might need comments.

And if these are wanted to have comments, let's use tje JSDoc style so they comments can be seen nicely in the IDE when using the interface.

E.g:
```
/** Navigation title */
navTitle: string;
```

File: src/routes/JobOpportunity/JobOpportunity.tsx (hunk 2 above)
Comment (juhaniko) [reply src/routes/JobOpportunity/JobOpportunity.tsx:9]: I'm not sure if the compiler does it automatically, but I'm under the assumption that this is more efficient way of importing types/interfaces, as they get omitted from the final bundle.

File: src/components/OpportunityDetails/OpportunityDetails.tsx (hunk 3 above)
Comment (juhaniko) [reply src/components/OpportunityDetails/OpportunityDetails.tsx]: True, these are pretty self explanatory. I removed the comments and renamed `dev` -> `showInDevOnly` to make more clear

File: src/routes/JobOpportunity/JobOpportunity.tsx (hunk 2 above)
Comment (ketsappi) [reply src/routes/JobOpportunity/JobOpportunity.tsx:9]: Yeah, it seems beneficial to explicitly tell the `type` after I read more about this.

It is also possible to use the following way, but it seems to have few gotchas so in general not recommended IIUC.
```tsx
import type { LoaderData } from './loader';
```



**So let's use like you already had done** 👍

Comment (juhaniko) [reply src/routes/JobOpportunity/JobOpportunity.tsx:9]: The mentioned approach won't work if a file exports mixed content, not just types. So you couldn't do `import type { FooType, FooComponent }`, but `import { type FooType, FooComponent }` works.

Comment (ketsappi) [reply src/routes/JobOpportunity/JobOpportunity.tsx:9]: Yeah, `import type` is only for types.

And the gotchas I mentioned, which in this particular case does not matter:

>Class Declarations: When importing a class using `import type`, you can’t use it to create instances or access static members

>Enums: Importing an enum with `import type` means you can't use its members as values

**Pull request merged**